### PR TITLE
add verify phone number

### PR DESCRIPTION
**Pull request by cedvdb**

### CODE REVIEW

**Comment by atn832** [comment]
Thanks for the PR @cedvdb. Would you mind implementing a unit test to demonstrate how the feature works?

**Comment by cedvdb** [comment]
done

File: test/firebase_auth_mocks_test.dart [hunk 1]
```diff
@@ -99,6 +101,23 @@ void main() {
       expect(auth.userChanges(), emitsInOrder([null, isA<User>()]));
       expect(user.isAnonymous, isTrue);
     });
+
+    test('should send sms code to user', () async {
+      final auth = MockFirebaseAuth();
+      final codeSent = Completer<bool>();
+      // currently does not auto retrieve the sms code like it does on Android
+      // it is left to the user to use the code to sign in, so we encompass
+      // every platform
```
Comment (atn832) [inline test/firebase_auth_mocks_test.dart:110]: Thanks for the example and comment. Reading https://firebase.flutter.dev/docs/auth/phone/#codesent was helpful too.

**Comment by atn832** [comment]
Just published your implementation on https://pub.dev/packages/firebase_auth_mocks/changelog#084. Thank you!

**Pull request merged**